### PR TITLE
feat: add tooltipProps to menu item

### LINKED ISSUE #11252: Allow Menu.Item rendering without Tooltip
- [X ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
Currently the only way to render menu item is using it inside Tooltip. This brings a problem, where when hovering over multiple items a lot of dom modifications have to be done for the tooltip. On slower devices this causes significant cpu usage. 

### What does the proposed API look like?
```javascript
import React from 'react';
import { Item } from 'rc-menu';
import { Menu, Tooltip } from 'antd';

/* 
  Because regular ant design component forces to use tooltips,
  which can result is high cpu usage for dom updates
 */
export class MenuItem extends Menu.Item {
  tooltipView(props) {
    const { inlineCollapsed } = this.context;
    return (
      <Tooltip
        title={inlineCollapsed && props.level === 1 ? props.children : ''}
        placement="right"
        overlayClassName={`${props.rootPrefixCls}-inline-collapsed-tooltip`}
      >
        <Item {...props} ref={this.saveMenuItem} />
      </Tooltip>
    );
  }

  itemView(props) {
    return <Item {...props} ref={this.saveMenuItem} />;
  }

  render() {
    const { props } = this;
    if (props.withTooltip) {
      return this.tooltipView(props);
    }
    return this.itemView(props);
  }
}
```

### PR DESCRIPTION
**Pull request by shanehofstetter**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #11252

### 💡 Background and solution

- Add tooltipProps to Menu.Item API to allow customization of the tooltip displayed on collapsed menu items

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add tooltipProps to MenuItem properties  |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/menu/index.en-US.md](https://github.com/shanehofstetter/ant-design/blob/feature-menu-item-tooltip-props/components/menu/index.en-US.md)

### CODE REVIEW

File: components/menu/index.en-US.md [hunk 1]
```diff
@@ -52,11 +52,12 @@ More layouts with navigation: [Layout](/components/layout).
...
-| -------- | ------------------------------------ | ------- | ------------- | ------- |
-| disabled | Whether menu item is disabled        | boolean | false         |         |
-| key      | Unique ID of the menu item           | string  |               |         |
-| title    | Set display title for collapsed item | string  |               | 3.11.0  |
+| Param        | Description                                             | Type                                | Default value | Version |
+| ------------ | ------------------------------------------------------- | ----------------------------------- | ------------- | ------- |
+| disabled     | Whether menu item is disabled                           | boolean                             | false         |         |
+| key          | Unique ID of the menu item                              | string                              |               |         |
+| title        | Set display title for collapsed item                    | string                              |               | 3.11.0  |
+| tooltipProps | The tooltip props                                       | [TooltipProps](/components/tooltip) |               |         |
```
Comment (afc163) [inline components/menu/index.en-US.md]: Don't format markdown, campat table format is better.

File: components/menu/index.en-US.md [hunk 2]
```diff
@@ -26,48 +26,49 @@ More layouts with navigation: [Layout](/components/layout).
...
-| -------- | ------------------------------------ | ------- | ------------- | ------- |
-| disabled | Whether menu item is disabled        | boolean | false         |         |
-| key      | Unique ID of the menu item           | string  |               |         |
-| title    | Set display title for collapsed item | string  |               | 3.11.0  |
+| Param | Description | Type | Default value | Version |
+| --- | --- | --- | --- | --- |
+| disabled | Whether menu item is disabled | boolean | false |  |
+| key | Unique ID of the menu item | string |  |  |
+| title | Set display title for collapsed item | string |  | 3.11.0 |
+| tooltipProps | The tooltip props | [TooltipProps](/components/tooltip) |  |  |
```
Comment (afc163) [inline components/menu/index.en-US.md]:
```suggestion
| tooltipProps | The tooltip props | [TooltipProps](/components/tooltip) |  | 3.25.0 |
```

File: components/menu/index.en-US.md [hunk 3]
```diff
@@ -26,48 +26,49 @@ More layouts with navigation: [Layout](/components/layout).
...
-| -------- | ------------------------------------ | ------- | ------------- | ------- |
-| disabled | Whether menu item is disabled        | boolean | false         |         |
-| key      | Unique ID of the menu item           | string  |               |         |
-| title    | Set display title for collapsed item | string  |               | 3.11.0  |
+| Param | Description | Type | Default value | Version |
+| --- | --- | --- | --- | --- |
+| disabled | Whether menu item is disabled | boolean | false |  |
+| key | Unique ID of the menu item | string |  |  |
+| title | Set display title for collapsed item | string |  | 3.11.0 |
+| tooltipProps | The tooltip props | [TooltipProps](/components/tooltip) |  | 3.25.0 |
```
Comment (afc163) [inline components/menu/index.en-US.md]: Could you help update Chinese API too, `The tooltip props` could be translated to `自定义收缩菜单的 tooltip 属性`

**Review by afc163 — APPROVED** [review]

**Comment by shanehofstetter** [comment]
@zombieJ @afc163 Just curious, why did you close this?

**Comment by yoyo837** [comment]
May be for the upcoming v4 beta, PR cleanup.

**Comment by afc163** [comment]
@zombieJ You should leave a message when closing PR.

**Pull request closed without merging**